[Fix] install.sh: Force remote name of cloned repo to be 'origin'

0cjs opened the pull request:
See the commit message(s) for details. (I avoid duplicating them here because the commit messages are being updated as we review.)

Feel free to change the commit message (or anything else) as necessary. The `[Fix]` prefix is not mentioned in the ["rest of the conventions"](https://gist.github.com/ljharb/772b0334387a4bee89af24183114b3c7) for commit messages, but seems to be what is used for similar things in previous commits.

This has been tested manually on my personal machine, where I use `clone.defaultRemoteName`. The automated tests against the current head of `master` (`make TEST_SUITE=fast test-bash`) had some issues for me, including what looked like spurious failures; I assume that this is just my configuration.

The CI tests below also appear to have unrelated issues, giving e.g. `sed: /etc/apt/sources.list: No such file or directory` on WSL.

ljharb commented:
The WSL changes are known; and the v0.40.0 failures are expected, so indeed it’s nothing you need to worry about.

0cjs commented:
Ok, I've started looking at adding a test for this, and I've run into a few issues. Some of these are no doubt due to me being unfamiliar with the code, but I'm documenting them here for posterity in case you want to consider mitigating any of them.

1. I noticed you make a Docker image available for development, so I tried to use that so I could avoid any local configuration issues. Unfortunately it didn't build. I fixed this, fixed a couple of minor errors in the docs, and did a bit of documentation cleanup, and filed this as PR #3657. All further notes below are from running things inside that container.

2. From README §Running Tests, `npm install` gives a vulnerabilities warning: "17 vulnerabilities (9 moderate, 7 high, 1 critical)". (I just ignored this.)

3. From the same section, `npm run test/fast` does not work:

        nvm@nvm-dev:~/.nvm$ \
            npm run test/fast

        > nvm@0.40.3 test/fast
        > shell=$(basename -- $(ps -o comm= $(ps -o ppid= -p $PPID)) | sed 's/^-//'); env -i TERM="$TERM" bash -lc "make TEST_SUITE=fast test-$shell"

        /usr/bin/env: 'node': No such file or directory
        Makefile:15: *** Did you forget to run `npm install` after cloning the repo? At least one of the required supporting utilities not found: urchin replace semver.  Stop.
        nvm@nvm-dev:~/.nvm$

   Node does seem to be available; `node --version` and `/usr/bin/env node --version` both give me "v24.8.0".

4. Moving on to just trying to run the `install_nvm_from_git` test script directly, running it from the root of the repo doesn't work:

        nvm@nvm-dev:~/.nvm$ \
            ./test/install_script/install_nvm_from_git
        ./test/install_script/install_nvm_from_git: 16: .: cannot open ../../install.sh: No such file

5. Running `install_nvm_from_git` from its own directory doesn't work:

        nvm@nvm-dev:~/.nvm/test/install_script$ \
            ./install_nvm_from_git
        Error: the install instructions explicitly say to pipe the install script to `bash`; please follow them

6. Oddly, running `bash` on that script does seem to work:

        nvm@nvm-dev:~/.nvm/test/install_script$ \
            bash ./install_nvm_from_git >../../z.log 2>&1 ; echo exitcode=$?
        exitcode=0
        # (Completes after about 45 seconds.)
        nvm@nvm-dev:~/.nvm/test/install_script$ \
            wc -l ../../z.log
        1717 ../../z.log

   So after about 45 seconds it completes without failure, and with a couple thousand lines of output, most of which looks like traces from the shell `-x` option.

   Note that running `bash test/install_script/install_nvm_from_git` from the root directory doesn't work; it also fails with `../../install.sh: No such file or directory`.

So am I correct that this script is intended to be run as follows?

    cd test/install_script/
    bash install_nvm_from_git >../../z.log 2>&1; echo exitcode=$?

I'll look through the script and give you further notes in the next comment.

0cjs commented:
So what are we currently waiting on in order to get this fix merged?

0cjs commented:
@ljharb Ping again; just wondering what's holding this up.
